### PR TITLE
Improve performances

### PR DESCRIPTION
**Pull request by BoboTiG**
Quick benchmarks (first line is the original code, second is the patched one):

```shell
$ python -m timeit -n 1000000 -s "from hexbytes import HexBytes" "HexBytes(0)"
1000000 loops, best of 5: 1.260 usec per loop
1000000 loops, best of 5: 0.875 nsec per loop (-30.5 %)

$ python -m timeit -n 1000000 -s "from hexbytes import HexBytes" "HexBytes('0')"
1000000 loops, best of 5: 883 nsec per loop
1000000 loops, best of 5: 760 nsec per loop (-13.9 %)

$ python -m timeit -n 1000000 -s "from hexbytes import HexBytes" "HexBytes(False)"
1000000 loops, best of 5: 540 nsec per loop
1000000 loops, best of 5: 447 nsec per loop (-17.2 %)
```

## What was wrong?

Nothing.

## How was it fixed?

Gut feeling.

I had to bypass mypy errors as they are not revelant.

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [x] Clean up commit history

[//]: # (For important changes that should go into the release notes please add a newsfragment file as explained here: https://github.com/ethereum/hexbytes/blob/master/newsfragments/README.md)

[//]: # (See: https://hexbytes.readthedocs.io/en/latest/contributing.html#pull-requests)
- [ ] Add entry to the [release notes](https://github.com/ethereum/hexbytes/blob/master/newsfragments/README.md)

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

**Comment by carver** [comment]
```
$ python -m timeit -n 1000000 -s "from hexbytes import HexBytes" "HexBytes(0)"
1000000 loops, best of 5: 1.260 usec per loop
1000000 loops, best of 5: 0.875 nsec per loop (-30.5 %)
```

I assume that `usec` is wrong, or it got >1000x faster? 😆

File: hexbytes/_utils.py [hunk 1]
```diff
@@ -11,27 +11,26 @@ def to_bytes(val: Union[bool, bytearray, bytes, int, str]) -> bytes:
...
-    elif isinstance(val, bool):
-        return b"\x01" if val else b"\x00"
-    elif isinstance(val, int):
-        # Note that this int check must come after the bool check, because
-        #   isinstance(True, int) is True
-        if val < 0:
-            raise ValueError(f"Cannot convert negative integer {val} to bytes")
+    type_var = type(val)
+    if type_var is bytes:
+        return val  # type: ignore
```
Comment (carver) [inline hexbytes/_utils.py:16]: My concern with this approach is that it doesn't handle subclasses. It looks like `HexBytes(Hexbytes(0))` would raise a `TypeError`, for example

**Comment by BoboTiG** [comment]
> ```
> $ python -m timeit -n 1000000 -s "from hexbytes import HexBytes" "HexBytes(0)"
> 1000000 loops, best of 5: 1.260 usec per loop
> 1000000 loops, best of 5: 0.875 nsec per loop (-30.5 %)
> ```
> 
> I assume that `usec` is wrong, or it got >1000x faster? laughing

I tried again, and indeed, I did a "small" mistake:

```python
# From the master branch
$ python -m timeit -n 1000000 -s "from hexbytes import HexBytes" "HexBytes(0)"
1000000 loops, best of 5: 1.2 usec per loop

# From the branch of that PR
$ python -m timeit -n 1000000 -s "from hexbytes import HexBytes" "HexBytes(0)"
1000000 loops, best of 5: 906 nsec per loop
```

File: hexbytes/_utils.py (hunk 1 above)
Comment (BoboTiG) [reply hexbytes/_utils.py:16]: Hm maybe the code is good enough as-is :+1:

**Pull request closed without merging**